test/gtest/core/test_mc_reduce.h

```diff
@@ -274,43 +282,26 @@ class test_mc_reduce : public testing::Test {
-                                           ReductionTest<UCC_DT_INT32, lor>,
-                                           ReductionTest<UCC_DT_INT64, lor>,
-                                           ReductionTest<UCC_DT_INT16, bor>,
-                                           ReductionTest<UCC_DT_INT32, bor>,
-                                           ReductionTest<UCC_DT_INT64, bor>,
-                                           ReductionTest<UCC_DT_INT16, lxor>,
-                                           ReductionTest<UCC_DT_INT32, lxor>,
-                                           ReductionTest<UCC_DT_INT64, lxor>,
-                                           ReductionTest<UCC_DT_INT16, bxor>,
-                                           ReductionTest<UCC_DT_INT32, bxor>,
-                                           ReductionTest<UCC_DT_INT64, bxor>,
-                                           ReductionTest<UCC_DT_FLOAT32, max>,
-                                           ReductionTest<UCC_DT_FLOAT64, max>,
-                                           ReductionTest<UCC_DT_FLOAT32, min>,
-                                           ReductionTest<UCC_DT_FLOAT64, min>,
-                                           ReductionTest<UCC_DT_FLOAT32, sum>,
-                                           ReductionTest<UCC_DT_FLOAT64, sum>,
-                                           ReductionTest<UCC_DT_FLOAT32, prod>,
-                                           ReductionTest<UCC_DT_FLOAT64, prod>>;
+using ReductionTypesOps = ::testing::Types<
+    ReductionTest<UCC_DT_INT16, max>, ReductionTest<UCC_DT_INT32, max>,
+    ReductionTest<UCC_DT_INT64, max>, ReductionTest<UCC_DT_INT16, min>,
+    ReductionTest<UCC_DT_INT32, min>, ReductionTest<UCC_DT_INT64, min>,
+    ReductionTest<UCC_DT_INT16, sum>, ReductionTest<UCC_DT_INT32, sum>,
+    ReductionTest<UCC_DT_INT64, sum>, ReductionTest<UCC_DT_INT16, prod>,
+    ReductionTest<UCC_DT_INT32, prod>, ReductionTest<UCC_DT_INT64, prod>,
+    ReductionTest<UCC_DT_INT16, land>, ReductionTest<UCC_DT_INT32, land>,
+    ReductionTest<UCC_DT_INT64, land>, ReductionTest<UCC_DT_INT16, band>,
+    ReductionTest<UCC_DT_INT32, band>, ReductionTest<UCC_DT_INT64, band>,
+    ReductionTest<UCC_DT_INT16, lor>, ReductionTest<UCC_DT_INT32, lor>,
+    ReductionTest<UCC_DT_INT64, lor>, ReductionTest<UCC_DT_INT16, bor>,
+    ReductionTest<UCC_DT_INT32, bor>, ReductionTest<UCC_DT_INT64, bor>,
+    ReductionTest<UCC_DT_INT16, lxor>, ReductionTest<UCC_DT_INT32, lxor>,
+    ReductionTest<UCC_DT_INT64, lxor>, ReductionTest<UCC_DT_INT16, bxor>,
+    ReductionTest<UCC_DT_INT32, bxor>, ReductionTest<UCC_DT_INT64, bxor>,
+    ReductionTest<UCC_DT_FLOAT32, max>, ReductionTest<UCC_DT_FLOAT64, max>,
+    ReductionTest<UCC_DT_FLOAT32, min>, ReductionTest<UCC_DT_FLOAT64, min>,
+    ReductionTest<UCC_DT_FLOAT32, sum>, ReductionTest<UCC_DT_FLOAT64, sum>,
+    ReductionTest<UCC_DT_FLOAT32, avg>, ReductionTest<UCC_DT_FLOAT64, avg>,
```

Comment:
what test are you running here? It doesn't check avg instead it does sum

Reply:
After a call with Sergei - this list is used for collective & mc reduce tests, it was clarified in mc reduce API that avg == sum.